mc2/controllers/docker/tests/test_docker_controller.py

```diff
@@ -2,10 +2,162 @@
 import responses
 from django.conf import settings
 from django.contrib.auth.models import User
+from hypothesis import given
+from hypothesis.extra.django import TestCase
+from hypothesis.strategies import text, random_module, lists, just
+
+from mc2.controllers.docker.tests.hypothesis_helper import (
+    models, DEFAULT_VALUE)
+
+from mc2.controllers.base.models import EnvVariable, MarathonLabel
 from mc2.controllers.base.tests.base import ControllerBaseTestCase
 from mc2.controllers.docker.models import DockerController
 
 
+def add_envvars(controller):
+    envvars = lists(models(EnvVariable, controller=just(controller)))
+    return envvars.map(lambda _: controller)
+
+
+def add_labels(controller):
+    labels = lists(models(MarathonLabel, controller=just(controller)))
+    return labels.map(lambda _: controller)
+
+
+def docker_controller(with_envvars=True, with_labels=True, **kw):
```

Comment:
A `Controller` (which I think has the wrong name) is a thing that manages a marathon job/app/whatever. A `DockerController` is one that manages an arbitrary docker container rather than something more specialised.